File: client-scalaz/src/main/scala/org/openapitools/client/api/Vehicle.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Vehicle._

case class Vehicle (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
name: Option[String],
description: Option[String],
vehicleType: Option[VehicleType],
vin: Option[String],
licensePlateNumber: Option[String],
externalId: Option[String],
hub: Option[ServiceHub],
programs: Option[List[Program]],
seatQuantity: Option[Long],
picture: Option[Asset],
markerIcon: Option[Asset],
trackingColor: Option[String],
belongsTo: Option[Account],
zone: Option[Region],
notes: Option[List[Note]],
noteCount: Option[Long],
inUse: Option[Boolean],
contentName: Option[String])

object Vehicle {
  import DateTimeCodecs._

  implicit val VehicleCodecJson: CodecJson[Vehicle] = CodecJson.derive[Vehicle]
  implicit val VehicleDecoder: EntityDecoder[Vehicle] = jsonOf[Vehicle]
  implicit val VehicleEncoder: EntityEncoder[Vehicle] = jsonEncoderOf[Vehicle]
}

File: client-clojure/src/sirqul_io_t_platform/specs/vehicle.clj
(ns sirqul-io-t-platform.specs.vehicle
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.vehicle-type :refer :all]
            [sirqul-io-t-platform.specs.service-hub :refer :all]
            [sirqul-io-t-platform.specs.program :refer :all]
            [sirqul-io-t-platform.specs.asset :refer :all]
            [sirqul-io-t-platform.specs.asset :refer :all]
            [sirqul-io-t-platform.specs.account :refer :all]
            [sirqul-io-t-platform.specs.region :refer :all]
            [sirqul-io-t-platform.specs.note :refer :all]
            )
  (:import (java.io File)))


(def vehicle-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :name) string?
   (ds/opt :description) string?
   (ds/opt :vehicleType) vehicle-type-spec
   (ds/opt :vin) string?
   (ds/opt :licensePlateNumber) string?
   (ds/opt :externalId) string?
   (ds/opt :hub) service-hub-spec
   (ds/opt :programs) (s/coll-of program-spec)
   (ds/opt :seatQuantity) int?
   (ds/opt :picture) asset-spec
   (ds/opt :markerIcon) asset-spec
   (ds/opt :trackingColor) string?
   (ds/opt :belongsTo) account-spec
   (ds/opt :zone) region-spec
   (ds/opt :notes) (s/coll-of note-spec)
   (ds/opt :noteCount) int?
   (ds/opt :inUse) boolean?
   (ds/opt :contentName) string?
   })

(def vehicle-spec
  (ds/spec
    {:name ::vehicle
     :spec vehicle-data}))
